pkg/controller/discovery/web/ns.go

```diff
@@ -117,29 +198,68 @@ func (h NsHandler) List(ctx *gin.Context) {
 //
 // Get a specific namespace on a cluster.
 func (h NsHandler) Get(ctx *gin.Context) {
-	status := h.Prepare(ctx)
-	if status != http.StatusOK {
-		ctx.Status(status)
-		return
+	ctx.Status(http.StatusMethodNotAllowed)
+}
+
+//
+// Has read access.
+func (h *NsHandler) hasRead(ns string) (bool, error) {
+	return h.rbac.Allow(&auth.Review{
+		Resource: auth.Namespace,
+		Verb:     auth.GET,
+		Name:     ns,
+	})
+}
+
+//
+// Has admin access.
+func (h *NsHandler) hasAdmin(ns string) (bool, error) {
+	return h.rbac.Allow(&auth.Review{
+		Group:     migapi.SchemeGroupVersion.Group,
+		Namespace: ns,
+		Resource:  auth.ALL,
+		Verb:      auth.ALL,
+	})
+}
+
+//
+// Has basic MIG user access.
+func (h *NsHandler) hasBasic(ns string) (bool, error) {
+	allowed, err := h.hasAdmin(ns)
```

Comment:
uhm I am confused, why does this function call `hasAdmin`? looks like the non-admin user will always return on line https://github.com/konveyor/mig-controller/pull/563/files#diff-e12f082564b41558aa0342877bc3fd84R231 when they should be going through checks on https://github.com/konveyor/mig-controller/pull/563/files#diff-e12f082564b41558aa0342877bc3fd84R236

Reply:
See the conversation I just "unresolved" above

Reply:
The SAR checks are very expensive in quantity.  This is an optimization for cases where the user is a cluster-admin and we're checking 2k namespaces (which can require 4k SAR).  For admins, this is intended to provided a x3 reduction.  For non-admins it adds one more check but the basic check will likely reject on the first SAR.

Reply:
The discovery service is authorizing the REST API and must be low latency.  This would not be necessary if we could agree on a balance between performance and certainty.  IMHO, the goal is not to provide an absolute guarantee the user has ALL necessary permission but instead a list of namespaces based on a quick estimate.  If the user has "*" on MigPlan the will most certainly have permission for everything else.  In any case, the downside to being wrong is they get an error message in the UI when they don't have permission.